From 03aa928deb75e977ab723f075e4df611f1350ffc Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Wed, 2 Mar 2022 13:43:37 +1100
Subject: [PATCH 5/5] avoid building static libs on windows when
 BUILD_SHARED_LIBS=ON

except all flags_* libraries, for which DLL builds on windows
are not supported

To do this, expand list of ABSL_INTERNAL_DLL_TARGETS; this
is due to how the logic of absl_internal_dll_contains checks
whether additional static libraries should be built, compare
https://github.com/abseil/abseil-cpp/blob/20211102.0/CMake/AbseilHelpers.cmake#L112-L132
---
 CMake/AbseilDll.cmake | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/CMake/AbseilDll.cmake b/CMake/AbseilDll.cmake
index 2c7f1628..18f21f67 100644
--- a/CMake/AbseilDll.cmake
+++ b/CMake/AbseilDll.cmake
@@ -366,6 +366,11 @@ set(ABSL_INTERNAL_DLL_TARGETS
   "container_common"
   "container_memory"
   "cord"
+  "cord_internal"
+  "cordz_functions"
+  "cordz_handle"
+  "cordz_info"
+  "cordz_sample_token"
   "core_headers"
   "counting_allocator"
   "debugging"
@@ -396,6 +401,7 @@ set(ABSL_INTERNAL_DLL_TARGETS
   "leak_check"
   "leak_check_disable"
   "log_severity"
+  "low_level_hash"
   "malloc_internal"
   "memory"
   "meta"
@@ -410,6 +416,7 @@ set(ABSL_INTERNAL_DLL_TARGETS
   "random_bit_gen_ref"
   "random_distributions"
   "random_internal_distribution_caller"
+  "random_internal_distribution_test_util"
   "random_internal_distributions"
   "random_internal_explicit_seed_seq"
   "random_internal_fast_uniform_bits"
@@ -445,6 +452,8 @@ set(ABSL_INTERNAL_DLL_TARGETS
   "stack_consumption"
   "stacktrace"
   "status"
+  "statusor"
+  "strerror"
   "str_format"
   "str_format_internal"
   "strings"
-- 
2.37.0.windows.1

